commit 2207162ab32f901e2504e357da32e09f8eee1b92 (refs/changes/41/5241/5)
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2018-12-14T18:01:09+00:00 (10 months ago)
    
    TRITON-1034 Provisioning a VM with `do_not_inventory` set causes the workflow job to time out at putVm
    Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
    Approved by: Todd Whiteman <todd.whiteman@joyent.com>

diff --git a/lib/workflows/job-common.js b/lib/workflows/job-common.js
index 0a91129..5b9f560 100644
--- a/lib/workflows/job-common.js
+++ b/lib/workflows/job-common.js
@@ -304,6 +304,12 @@ function pollTask(job, cb) {
 function putVm(job, cb) {
     var vmapi;
 
+    if (job.params.payload.do_not_inventory) {
+        cb(null,
+            'VM has do_not_inventory set, no need to putVm to VMAPI');
+        return;
+    }
+
     /*
      * Checks (polls) the state of a machine in VMAPI. It is used for provisions
      * and VM actions such as reboot and shutdown.
